### PR TITLE
Add nix expression to get working development environment

### PR DESCRIPTION
**Pull request by tolbrino**
This enables users of [nix](https://nixos.org/nix/) (e.g. on [NixOS](https://nixos.org/)) to start a working
development environment by running `nix-shell` in the top folder.

### CODE REVIEW

**Comment by tolbrino** [comment]
There is a [docker image](https://hub.docker.com/r/nixorg/nix) which could be used to easily test it. I wanted to have the PR up as a reference even if it doesn't get merged.

**Comment by tolbrino** [comment]
@venimus @dincho I added a build check to CircleCI which works nicely. Re-reviews needed.

File: .circleci/config.yml [hunk 1]
```diff
@@ -128,6 +131,12 @@ references:
     restore_cache:
       key: *build_otp21_cache_key
 
+  build_nix_cache_key: &build_nix_cache_key build-nix-cache-v1-{{ .Revision }}
+  restore_build_nix_cache: &restore_build_nix_cache
+    restore_cache:
+      key: *build_nix_cache_key
+
+
```
Comment (venimus) [inline .circleci/config.yml]: extra nl

**Comment by tolbrino** [comment]
@dincho More thoughts?

**Comment by tolbrino** [comment]
> Not sure if we need to run this on each commit (and waste resources) as the only expected fail point is the `default.nix` I guess?
> 
> Perhaps is more wise to run it on master only ?

@dincho @venimus I included your feedback.

**Pull request merged**